3.14 release post (#323)
ember-learn/ember-blog: source/2019-11-01-ember-3-14-released.md

You can read more about our general release process here:

- [Release Dashboard](http://emberjs.com/builds/)
- [The Ember Release Cycle](http://emberjs.com/blog/2013/09/06/new-ember-release-process.html)
- [The Ember Project](http://emberjs.com/blog/2015/06/16/ember-project-at-2-0.html)
- [Ember LTS Releases](http://emberjs.com/blog/2016/02/25/announcing-embers-first-lts.html)

---

## The Octane Preview, Continued

As described in yesterday's post, [Octane Release Update](/2019/10/31/octane-release-update.html), Octane will continue to be opt-in in Ember 3.14. By continuing the Octane preview period in 3.14, we are allowing more time for Octane to be polished before it becomes the default Ember experience.

Although Octane is still opt-in in 3.14, all Octane features are stable public APIs with our normal SemVer guarantees. If you build an app using APIs in the Octane preview, it will continue to work in future versions.

---

> With the exception of an updated release of `@glimmer/component`, the steps to try out the Octane preview remain the same as they were in 3.13. You can try out the Octane preview by performing a few steps:

Reviewer: ```suggestion
With the exception of an updated release of `@glimmer/component`, the steps to try out the Octane preview remain the same as in 3.13. You can try out the Octane preview by doing the following:
```

Add the following packages if they aren't already present at this version:

```js
npm install --save-dev @ember/edition-utils@^1.1.1
npm install --save-dev @glimmer/component@^1.0.0-beta.1
```

Disable legacy behavior by setting the following feature flags:

```bash

> ember feature:disable jquery-integration

Reviewer 2: I thought disabling jQuery was no longer a requirement for opting into Octane? From https://blog.emberjs.com/2019/10/31/octane-release-update.html

> Because there is no technical reason to couple the removal of jQuery integration with Octane, we decided not to require Octane apps to disable jQuery integration. This requirement change is implemented as of the 3.14.0 release.

Reviewer: @patocallaghan thanks - you're correct! I just carried this over from the 3.13 post.

A template such as the following can now be refactored from using `this.model`:

```hbs
{{!-- The model for this route is the current user --}}

<div>
  Hi <img src="{{this.model.profileImage}}" alt="{{this.model.name}}'s profile picture"> {{this.model.name}},
  this is a valid Ember template!
</div>

{{#if this.model.isAdmin}}
  <div>Remember, with great power comes great responsibility!</div>
{{/if}}
```

To use `@model`:

```hbs
{{!-- The model for this route is the current user --}}

<div>
  Hi <img src="{{@model.profileImage}}" alt="{{@model.name}}'s profile picture"> {{@model.name}},
  this is a valid Ember template!
</div>

{{#if @model.isAdmin}}
  <div>Remember, with great power comes great responsibility!</div>
{{/if}}
```

The guides will soon be updated to teach the `@model` form first-and-foremost.

#### Deprecations (0)

There are no new deprecations in Ember.js 3.14.

For more details on changes in Ember.js 3.14, please review the [Ember.js 3.14.0 release page](https://github.com/emberjs/ember.js/releases/tag/v3.14.0).

---

## Ember Data

Ember Data is the official data persistence library for Ember.js applications.

> ### Changes in Ember Data 3.14

Reviewer 3: @igorT @dgeb no changes at all to Ember Data?

Reviewer: I think there will be. I'll add the ember data + cli changes once they've been released. Currently, these sections are just empty boilerplate.

Author: @dgeb Any updates? Or do you need any support?

Reviewer: > Any updates? Or do you need any support?

@kennethlarsen ember cli was released today and I just finished updating this post with both data + cli changes. A review would be much appreciated!
